test/LibraryManager.IntegrationTest/VisualStudioLibmanHostTests.cs

```diff
@@ -38,13 +41,51 @@ protected override void DoHostTestInitialize()
 
             Solution.Open(_solutionPath);
             Solution.WaitForFullyLoaded(); // This will get modified after bug 627108 get fixed
+
+            string pathToLibmanFile = Path.Combine(SolutionRootPath, _projectName, _libman);
+            _initialLibmanFileContent = File.ReadAllText(pathToLibmanFile);
         }
 
         protected override void DoHostTestCleanup()
         {
+            ProjectTestExtension webProject = Solution[_projectName];
+            ProjectItemTestExtension libmanConfig = webProject[_libman];
+
+            if (libmanConfig != null)
+            {
+                CleanClientSideLibraries();
+
+                libmanConfig.Open();
+                Editor.Selection.SelectAll();
+                Editor.KeyboardCommands.Delete();
+                Editor.Edit.InsertTextInBuffer(_initialLibmanFileContent);
+
+                libmanConfig.Save();
+            }
+
             base.DoHostTestCleanup();
         }
 
+        private void CleanClientSideLibraries()
+        {
+            Guid guid = Guid.Parse("44ee7bda-abda-486e-a5fe-4dd3f4cefac1");
+            uint commandId = 0x0200;
+            SolutionExplorerItemTestExtension libmanConfigNode = SolutionExplorer.FindItemRecursive(_libman);
```

Comment:
>SolutionExplorerItemTestExtension libmanConfigNode = SolutionExplorer.FindItemRecursive(_libman); [](start = 12, length = 97)

This could be fragile if we ever add other projects to the test solution.  Maybe you should pass in `webProject[_libman]` from the caller instead of doing another search here?

It's not perfect, but makes this function more reusable in the future.

Reply:
Oh, I guess `webProject[_libman]` is a ProjectItemTestExtension, not a SolutionExplorerTestExtension.  Sigh...

---
In reply to: [208029266](https://github.com/aspnet/LibraryManager/pull/341#discussion_r208029266) [](ancestors = 208029266)

Reply:
Yes, I thought I could do that. But I don't see an option to Select(..) in ProjectItemTestExtension 

---
In reply to: [208029571](https://github.com/aspnet/LibraryManager/pull/341#discussion_r208029571) [](ancestors = 208029571,208029266)